### PR TITLE
INTYGFV-14489: Added support in Testability to create Ts-Diabetes v2 certificates.

### PR DESCRIPTION
**Pull request by ChrilleSigerhed**

### CODE REVIEW

File: support/src/main/java/se/inera/intyg/common/support/facade/util/ViewTextToolkit.java [hunk 1]
```diff
@@ -80,10 +80,10 @@ public static String multipleStringValues(String... value) {
             return NOT_SPECIFIED;
         }
 
-        return stringBuilder.toString().trim();
+        return text;
     }
 
-    public static String withComma(String... value) {
+    public static String multipleStringValuesWithComma(String... value) {
```
Comment (mww-sopra-steria) [inline support/src/main/java/se/inera/intyg/common/support/facade/util/ViewTextToolkit.java:86]: This method has "logic" that isn't covered by test. Maybe could add some test cases for handling null, trimming and empty?

File: ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v2/testability/TSTRK1031TestabilityTestDataDecorator.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import se.inera.intyg.common.support.facade.util.TestabilityUtlatandeTestDataProvider;
+import se.inera.intyg.common.support.model.InternalDate;
+import se.inera.intyg.common.ts_diabetes.v2.model.internal.BedomningKorkortstyp;
+import se.inera.intyg.common.ts_diabetes.v2.model.internal.IntygAvserKategori;
+import se.inera.intyg.common.ts_diabetes.v2.model.internal.TsDiabetesUtlatandeV2;
+import se.inera.intyg.common.ts_diabetes.v2.model.internal.Vardkontakt;
+import se.inera.intyg.common.ts_parent.codes.DiabetesKod;
+import se.inera.intyg.common.ts_parent.codes.IdKontrollKod;
+
+public class TSTRK1031TestabilityTestDataDecorator implements TestabilityUtlatandeTestDataProvider<TsDiabetesUtlatandeV2> {
```
Comment (mww-sopra-steria) [inline ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v2/testability/TSTRK1031TestabilityTestDataDecorator.java:32]: Rename this class TSTRK1031TestabilityUtlatandeTestDataProvider. Also for consistency I would suggest calling it TsDiabetes instead of using the Tstrk...

**Review by mww-sopra-steria — APPROVED** [review]
Looks great! Approved!

Two minor suggestions that you can take a look at before you decide to merge.

File: ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v2/testability/TSTRK1031TestabilityTestDataDecorator.java (hunk 2 above)
Comment (ChrilleSigerhed) [reply ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v2/testability/TSTRK1031TestabilityTestDataDecorator.java:32]: Done

File: support/src/main/java/se/inera/intyg/common/support/facade/util/ViewTextToolkit.java (hunk 1 above)
Comment (ChrilleSigerhed) [reply support/src/main/java/se/inera/intyg/common/support/facade/util/ViewTextToolkit.java:65]: added tests!
